crates/vm/src/arch/vm.rs

```diff
@@ -473,7 +486,13 @@ where
         if let Some(overridden_heights) = self.overridden_heights.as_ref() {
             segment.set_override_trace_heights(overridden_heights.clone());
         }
-        metrics_span("execute_time_ms", || segment.execute_from_pc(pc_start))?;
+        let mut vm_state = TracegenVmExecutionState::from_pc_and_memory_controller(
```

Comment:
this flow seems a bit weird - ok if it's just temporary until we get rid of or refactor memory controller

Reply:
yeah, it's temporary until i figure out how to get rid of memory controller owning the memory and timestamp